apps/browser/browserservice.cpp

```diff
@@ -106,7 +107,15 @@ void BrowserService::dumpMemoryInfo(const QString &fileName)
 
 bool BrowserService::isPrivileged() const
 {
-    IS_PRIVILEGED;
+    auto isPrivileged = [=] {
+        IS_PRIVILEGED();
```

Comment:
That "double semicolon" thing == what happens here. 'IS_PRIVILEGED();' expands to '... == "root";;' - which is harmless, but can also trigger empty statement warnings / stte. The canonical thing to do would be to enclose macro def in "do { multiple; statements; } while(0)" -> semicolon comes from where the macro is used, and as a bonus also things like "if (foo) MACRO(x);" behaves like one might expect when looking at the code.

On a closer look. What is this? Lambda method to work around a macro that looks like a predicate but actually does a return? I guess it works and static function / similar is out of the question due to calledFromDBus() method requiring some suitable object context?

Since this was just something that caught by eye, not what the PR was about: It is fine as it is, but if you wish to make places where the macro is used look more like regular code, one alternative might be to use statement expression (gnu extension) so that it evaluates into boolean value instead of doing a value return -> could avoid that lambda here / have return IS_PRIVILEGED() in that other place. Then again I'm not 100% sure if this c extensions also c++ ok and whether it is better / worse than what there already is depends on the reader...

```
#define IS_PRIVILEGED() ({\
    bool isPrivileged = true;\
    if (calledFromDBus()) {\
        uint pid = GET_PID(); \
        QFileInfo info(QString("/proc/%1").arg(pid)); \
        isPrivileged = info.group() == "privileged" || info.owner() == "root";\
    }\
    isPrivileged;\
})

```